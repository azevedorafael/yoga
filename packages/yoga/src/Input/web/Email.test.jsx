import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, Input } from '../..';

describe('<Input.Email />', () => {
  describe('Snapshots', () => {
    it('should match with default input', () => {
      const { container } = render(
        <ThemeProvider>
          <Input.Email />
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
