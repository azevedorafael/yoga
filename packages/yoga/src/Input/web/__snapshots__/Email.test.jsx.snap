// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input.Email /> Snapshots should match with default input 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
  border-radius: 8px;
  border: 1px solid #D3D3E2;
}

.c0:hover,
.c0:focus-within {
  border-color: #41414A;
}

.c0 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A6;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c0 svg:hover,
.c0 svg:focus {
  fill: #41414A;
}

.c0 {
  width: 312px;
}

.c1 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 40px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
}

.c1:focus {
  color: #41414A;
}

.c1:focus + label {
  top: -1px;
  left: 14px;
  padding-right: 4px;
  padding-left: 4px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c1:disabled {
  cursor: not-allowed;
  border-color: #D3D3E2;
  color: #D3D3E2;
}

.c1::-webkit-input-placeholder {
  color: #9898A6;
}

.c1::-moz-placeholder {
  color: #9898A6;
}

.c1:-ms-input-placeholder {
  color: #9898A6;
}

.c1::placeholder {
  color: #9898A6;
}

.c1[type="number"]::-webkit-outer-spin-button,
.c1[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2 {
  position: absolute;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  top: 24px;
  left: 16px;
  background-color: #FFFFFF;
  font-size: 14px;
  font-weight: 400;
  color: #9898A6;
  -webkit-transform: translateY(-40%);
  -ms-transform: translateY(-40%);
  transform: translateY(-40%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
}

<div>
  <div>
    <div
      class="c0"
    >
      <input
        class="c1"
        type="email"
        value=""
      />
      <label
        class="c2"
      >
        Label
      </label>
    </div>
  </div>
</div>
`;
